presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -787,6 +788,12 @@ protected RowExpression visitSubscriptExpression(SubscriptExpression node, Void
             RowExpression base = process(node.getBase(), context);
             RowExpression index = process(node.getIndex(), context);
 
+            //this block will handle row subscript, converts the ROW_CONSTRUCTOR with subscript to a DEREFERENCE expression
+            if (base instanceof SpecialFormExpression && base.getType() instanceof RowType) {
+                ConstantExpression pos = (ConstantExpression) index;
+                int i = toIntExact((Long) pos.getValue());
+                return specialForm(DEREFERENCE, getType(node), base, new ConstantExpression(i - 1, INTEGER));
```

Comment:
Presto cannot recognize `int` as primitive type. Make sure `i - 1` is of `long` type.